### PR TITLE
Updates leads primary company name after company name changed

### PR DESCRIPTION
**Pull request by kuzmany**
…mary

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Yes
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5652
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

This bugfix/feature has been sponsored by @Webmecanik

[//]: # ( Required: )
#### Description:
Contact's primary company name wasn't change after company edit.
Reported here https://github.com/mautic/mautic/issues/5652

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1.  Assign company to contact 
2.  Change company name
3. See contacts detail  and see old company name  (or see column company in leads table)

#### Steps to test this PR:
1.  Repeat all steps
2.  See company name was update

### CODE REVIEW

**Comment by kuzmany** [comment]
Label: Ready to test

**Comment by tristanbailey** [comment]
I have a clean Mautic 2.12.1 install, with this PR, I did see the company name change on the edit page, but not on the details for the Contact.

screen shot 2018-02-05 at 11 29 08
screen shot 2018-02-05 at 11 28 59

**Comment by kuzmany** [comment]
@tristanbailey fixed, please retest

**Comment by npracht** [comment]
@justinfortes could we add that as candidate for 2.13.0 knowing that this is a bugfix !?
Thanks !

**Comment by npracht** [comment]
Label: Pending test confirmation

**Comment by tristanbailey** [comment]
I will see if I can test this again this week, or anyone else obviously.

File: app/bundles/LeadBundle/Model/CompanyModel.php [hunk 1]
```diff
@@ -78,6 +78,7 @@ public function __construct(FieldModel $leadFieldModel, Session $session, EmailV
     public function saveEntity($entity, $unlock = true)
     {
         $this->setEntityDefaultValues($entity, 'company');
+        $this->getCompanyLeadRepository()->updateLeadsPrimaryCompanyName($entity);
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Model/CompanyModel.php]: This will also need to be applied to saveEntities for companies edited in batch via the API.

**Comment by kuzmany** [comment]
@escopecz  fixed and ready to test againg

**Pull request merged**